core/src/main/scala/com/databricks/labs/remorph/parsers/tsql/TSqlExpressionBuilder.scala

```diff
@@ -1,24 +1,38 @@
 package com.databricks.labs.remorph.parsers.tsql
 
+import com.databricks.labs.remorph.parsers.tsql.TSqlParser._
+import com.databricks.labs.remorph.parsers.{ParserCommon, intermediate => ir}
 import org.antlr.v4.runtime.Token
 import org.antlr.v4.runtime.tree.TerminalNode
 
-import com.databricks.labs.remorph.parsers.intermediate.{Column, Literal}
-import com.databricks.labs.remorph.parsers.tsql.TSqlParser._
-import com.databricks.labs.remorph.parsers.{IncompleteParser, ParserCommon, intermediate => ir}
+class TSqlExpressionBuilder extends TSqlParserBaseVisitor[ir.Expression] with ParserCommon {
 
-import scala.collection.JavaConverters._
+  // TODO: A lot of work here for things that are not just simple x.y.z
+  override def visitSelectListElem(ctx: TSqlParser.SelectListElemContext): ir.Expression =
+    ctx.expressionElem.accept(this)
 
-class TSqlExpressionBuilder
-    extends TSqlParserBaseVisitor[ir.Expression]
-    with ParserCommon
-    with IncompleteParser[ir.Expression] {
+  override def visitFullTableName(ctx: FullTableNameContext): ir.Literal = {
+    // Extract the components of the full table name, if they exist
+    val linkedServer = Option(ctx.linkedServer).map(_ => ctx.linkedServer.getText + ".")
+    val database = Option(ctx.database).map(_.getText)
+    val schema = Option(ctx.schema).map(_.getText)
+    val name = ctx.table.getText
 
-  protected override def wrapUnresolvedInput(unparsedInput: String): ir.UnresolvedExpression =
-    ir.UnresolvedExpression(unparsedInput)
+    val unparsedIdentifier = List(linkedServer, database, schema, Some(name)).flatten.mkString(".")
+    ir.Literal(string = Some(unparsedIdentifier))
+  }
 
-  override def visitFullColumnName(ctx: FullColumnNameContext): ir.Expression = {
-    Column(ctx.id.getText)
+  override def visitFullColumnName(ctx: FullColumnNameContext): ir.Column = {
+    val columnName = ctx.id.getText
+    val fullColumnName = ctx.fullTableName() match {
+      case null => columnName
+      case ft =>
+        ft.accept(this) match {
+          case nt: ir.Literal if nt.string.isDefined => nt.string.get + "." + columnName
+          case _ => columnName
+        }
+    }
```

Comment:
```suggestion
    val fullColumnName = Option(ctx.fullTableName())
      .map(_.accept(this))
      .collect{
          case nt: ir.Literal if nt.string.isDefined => nt.string.get + "." + columnName
       }
       .getOrElse(columnName) 
    }
```

nitpicking

Reply:
I am not sure that this was my code - I just moved it. However, I have changed it.